pkg/envoy/xds/server.go

```diff
@@ -321,7 +321,7 @@ func (s *Server) processRequestStream(ctx context.Context, streamLog *logrus.Ent
 			state := &typeStates[index]
 			watcher := s.watchers[typeURL]
 
-			nodeIP, err := IstioNodeToIP(req.GetNode())
+			nodeIP, err := IstioNodeToIP(req.GetNode().GetId())
```

Comment:
IstioNodeToIP is still returning an 'Ip'?

Reply:
Yes, the return value did not change here; it is the IP address that is by (Istio) convention embedded into the xDS Node Id.